src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs

```diff
@@ -267,13 +267,30 @@ void CollectionItemsSourceChanged(object sender, System.Collections.Specialized.
 				return;
 			}
 
+			// While Modifying the collection we should consider the ItemsUpdatingScrollMode to update the position
+			if (Carousel.ItemsUpdatingScrollMode == ItemsUpdatingScrollMode.KeepLastItemInView)
```

Comment:
Can simplify it:

```
if (Carousel.ItemsUpdatingScrollMode == ItemsUpdatingScrollMode.KeepLastItemInView)
    carouselPosition = (count == 0) ? 0 : count - 1;
else if (Carousel.ItemsUpdatingScrollMode == ItemsUpdatingScrollMode.KeepItemsInView)
    carouselPosition = 0;
```

Reply:
I think is better to keep as is to better understand the different positions we want

Reply:
Hi @jsuarezruiz , I agree with that. Keeping it as it is will provide better clarity in understanding the different positions we want to define.